tests/php/Control/SessionHandler/DatabaseSessionHandlerTest.php

```diff
@@ -0,0 +1,275 @@
+    {
+        return [
+            'deletes existing session' => [
+                'sessionID' => 'valid',
+            ],
+            'deletes expired session' => [
+                'sessionID' => 'expired',
+            ],
+            'no action for missing session' => [
+                'sessionID' => 'new-session'
+            ],
+        ];
+    }
+
+    #[DataProvider('provideDestroy')]
+    public function testDestroy(string $sessionID): void
+    {
+        $handler = new DatabaseSessionHandler();
+
+        $this->assertTrue($handler->destroy($sessionID));
+        $result = DB::query('SELECT "ID" FROM "_sessions" WHERE "ID" = \'' . $sessionID . '\'');
+        $this->assertSame(0, $result->numRecords());
+    }
+
+    public static function provideGc(): array
+    {
+        return [
+            'respect config' => [
+                'gcLifetime' => 100,
+                'configLifetime' => 500,
+                'sessionLifetimeMap' => [
+                    'session1' => 50,
+                    'session2' => 550,
+                    'session3' => 150,
+                    'session4' => 1000,
+                ],
+                'expectDeleted' => [
+                    'session2',
+                    'session4',
+                    'expired',
+                ],
+            ],
+            'fall back on gc' => [
+                'gcLifetime' => 100,
+                'configLifetime' => 0,
+                'sessionLifetimeMap' => [
+                    'session1' => 50,
+                    'session2' => 550,
+                    'session3' => 150,
+                    'session4' => 1000,
+                ],
+                'expectDeleted' => [
+                    'session2',
+                    'session3',
+                    'session4',
+                    'expired',
+                ],
+            ],
+        ];
+    }
+
+    #[DataProvider('provideGc')]
+    public function testGc(int $gcLifetime, int $configLifetime, array $sessionLifetimeMap, array $expectDeleted): void
+    {
+        ini_set('session.gc_maxlifetime', $gcLifetime);
+        Session::config()->set('timeout', $configLifetime);
+        $lifetime = ($configLifetime > 0) ? $configLifetime : $gcLifetime;
+        $handler = new DatabaseSessionHandler();
+
+        // Create the dummy files and set their modified time as appropriate
+        // Use symfony filesystem so that any subdirs are automatically created
+        $now = DBDatetime::now();
+        DBDatetime::set_mock_now($now);
+        // $lifeToDate represents how long ago the session was last written to
+        // so we need to calculate its expiry date based on how much lifetime is left.
+        foreach ($sessionLifetimeMap as $sessionID => $lifeToDate) {
+            $expiry = $now->getTimestamp() + ($lifetime - $lifeToDate);
+            DB::query(sprintf(
+                'INSERT INTO "_sessions" ("ID", "Data", "Expiry") VALUES (\'%s\', \'%s\', \'%s\')',
```

Comment:
```suggestion
        $tableName = DatabaseSessionHandler::config()->get('table_name');
        foreach ($sessionLifetimeMap as $sessionID => $lifeToDate) {
            $expiry = $now->getTimestamp() + ($lifetime - $lifeToDate);
            DB::query(sprintf(
                'INSERT INTO "' . $tableName . '" ("ID", "Data", "Expiry") VALUES (\'%s\', \'%s\', \'%s\')',
```

Reply:
Done (hard to see what's changed in the github diff but I assume it's just the table name)